### ISSUE TITLE
Make device description a clickable link

### ISSUE THREAD

**Issue by llamafilm** [netbox-community/netbox#15981]
### NetBox version

v3.7.3

### Feature type

Change to existing functionality

### Proposed functionality

When viewing the `/dcim/devices` webpage, I would like an option to make the `Description` field a hyperlink to the device page, just like the `ID` and `Name` columns behave now.

Implementation details are open for discussion.  It could be always on, or a configuration.py parameter, or a user preference.

### Use case

We don't use the `Name` field at all because the uniqueness constraints don't work for our use case.  All of our devices are "unnamed", so I like to hide that column while browsing the UI.  For now, I'm keeping the `ID` column visible so I can click the link to the device.  I would prefer to hide that column and click on the `Description` instead, because the ID number is meaningless for most users, and it's also requires more mouse precision to click the smaller link area.

Current view:
![image](https://github.com/netbox-community/netbox/assets/3991046/05d04683-eb4a-42e4-bc1d-6139c3aa94d5)

Desired change:
![image](https://github.com/netbox-community/netbox/assets/3991046/4f08e9e5-536d-48c4-ac1e-f909d233e16d)

This might be an edge case, but it would help a lot for usability in our situation and I hope it would be trivial to implement.

### External dependencies

none

**Comment by jeffgdotorg** [comment]
Good FR, thanks for writing it up.

**Comment by Julio-Oliveira-Encora** [comment]
Could you please assign it to me?

**Comment by jeremystretch** [comment]
I'm afraid the proposed change isn't tenable, for two reasons.

First, the description field is not intended to be used in this manner: It is meant to capture a brief bit of free-form text, and linking this text in most cases would be unnatural and confusing.

Second, most objects in NetBox have a description field, and linkifying only this one would be inconsistent with the others.

Maybe you could expand on why you're not using the name field for devices?

**Comment by llamafilm** [comment]
Our devices do not have unique names. For example, at one site I might have 100 devices called “AVR”. Within the context of a room (Location), this makes sense and is the easiest way for our technicians to refer to a device. When I need a unique identifier, I use the Device ID.

When I first started using Netbox, I tried to think of a naming convention to ensure uniqueness, but the only way to do that is to append information that’s already present in other fields, which makes it harder to read and more prone to mistakes. So I used the Description field as a “human friendly” label in the Netbox UI and as a label in Prometheus/Grafana, which is fed by Netbox.
Then I realized it’s possible to leave all devices unnamed which works great, except for this issue. 

Another option is to relax the requirement for unique names per site, but I think that has already been shot down before.
